src/openms/source/ANALYSIS/ID/BasicProteinInferenceAlgorithm.cpp

```diff
@@ -284,6 +327,28 @@ namespace OpenMS
       }
       //}
     }
+  }
+
+
+  void BasicProteinInferenceAlgorithm::updateProteinScores_(
+      std::unordered_map<std::string, std::pair<ProteinHit*, Size>>& acc_to_protein_hitP_and_count,
+      std::unordered_map<std::string, std::map<Int, PeptideHit*>>& best_pep,
+      bool pep_scores,
+      bool higher_better) const
+  {
+    //TODO Allow count as aggregation method -> i.e. set as protein score?
+
+    if (!higher_better && pep_scores)
+    {
+      higher_better = true; // We will convert the scores to PPs for multiplication
+    }
+
+    bool skip_count_annotation(param_.getValue("skip_count_annotation").toBool());
+
+    String agg_method_string(param_.getValue("score_aggregation_method").toString());
+
+    AggregationMethod aggregation_method = aggFromString_(agg_method_string);
+    const auto& aggregation_fun = aggFunFromEnum_(aggregation_method, higher_better);
```

Comment:
The lifetime extension is fine here I guess. I am Always unsure with temporaries